gridworld.py

```diff
@@ -2,17 +2,32 @@
 
 from environment import GridworldEnvironment
 from simulator import GridworldSimulator
+from mdp import MDP
 
 grid = np.array([
-        [0, 0, 0, 0, 1],
-        [0, 1, 1, 0, 1],
-        [0, 0, 'D', 0, 1],
-        [0, 1, 1, 0, 1],
-        [0, 0, 'S', 0, 1],
+    [0, 0, 0, 0, 1],
+    [0, 1, 1, 0, 1],
+    [0, 0, 'D', 0, 1],
+    [0, 1, 1, 0, 1],
+    [0, 0, 'S', 0, 1],
 ])
 Pe = 0.3
 initial_state = (0, 2)
 
 gridworld_environment = GridworldEnvironment(grid, Pe)
+
+# TODO: Find a better place to define this
+R = {}
+for state in gridworld_environment.S:
+    for action in gridworld_environment.A:
+        for next_state in gridworld_environment.S:
+            R.setdefault(state, {})
+            R[state].setdefault(action, {})
+            R[state][action][next_state] = 1 if grid[next_state] == 'D' or grid[next_state] == 'H' else 0
+
```

Comment:
This could be defined in the environment classes, right after S, A, P, and O. I'll push a commit later to do that